pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -1412,58 +1409,36 @@ func taintX509CA(tx *gorm.DB, trustDomainID string, publicKeyToTaint crypto.Publ
 	return nil
 }
 
-func revokeX509CA(tx *gorm.DB, trustDomainID string, publicKeyToRevoke crypto.PublicKey) error {
+func revokeX509CA(tx *gorm.DB, trustDomainID string, subjectKeyIDToRevoke string) error {
 	bundle, err := getBundle(tx, trustDomainID)
 	if err != nil {
 		return err
 	}
 
-	var taintedKeyFound bool
-	var taintedKeys []*common.X509TaintedKey
-
-	for _, eachTaintedKey := range bundle.X509TaintedKeys {
-		taintedKey, err := x509.ParsePKIXPublicKey(eachTaintedKey.PublicKey)
-		if err != nil {
-			return status.Errorf(codes.Internal, "failed to parse tainted Key: %v", err)
-		}
-
-		ok, err := cryptoutil.PublicKeyEqual(taintedKey, publicKeyToRevoke)
-		if err != nil {
-			return status.Errorf(codes.Internal, "failed to compare public key: %v", err)
-		}
-		if ok {
-			taintedKeyFound = true
-			continue
-		}
-
-		taintedKeys = append(taintedKeys, eachTaintedKey)
-	}
-
-	if !taintedKeyFound {
-		return status.Error(codes.InvalidArgument, "it is not possible to revoke an untainted root CA")
-	}
-	bundle.X509TaintedKeys = taintedKeys
-
-	// It is possible to keep bundles on journal, when there is no upstream authority,
-	// this code will be used to remove a CA bundle that is persisted on datastore,
-	// only in case it is found
+	keyFound := false
 	var rootCAs []*common.Certificate
 	for _, ca := range bundle.RootCas {
 		cert, err := x509.ParseCertificate(ca.DerBytes)
 		if err != nil {
 			return status.Errorf(codes.Internal, "failed to parse root CA: %v", err)
 		}
 
-		ok, err := cryptoutil.PublicKeyEqual(cert.PublicKey, publicKeyToRevoke)
-		if err != nil {
-			return status.Errorf(codes.Internal, "failed to compare public key: %v", err)
-		}
-
-		if ok {
+		caSubjectKeyID := x509util.SubjectKeyIDToString(cert.SubjectKeyId)
+		if subjectKeyIDToRevoke == caSubjectKeyID {
+			if !ca.TaintedKey {
+				return status.Error(codes.InvalidArgument, "it is not possible to revoke an untainted root CA")
+			}
+			keyFound = true
 			continue
 		}
+
 		rootCAs = append(rootCAs, ca)
 	}
+
+	if !keyFound {
+		return status.Error(codes.NotFound, "no root CA found with provided subject key ID")
+	}
+
 	bundle.RootCas = rootCAs
 
 	if _, err := updateBundle(tx, bundle, nil); err != nil {
```

Comment:
I think that we should increment bundle.SequenceNumber before calling updateBundle, otherwise SequenceNumber will remain the same.